spec/lib/object_serializer_relationship_serializer_proc_spec.rb

```diff
@@ -27,6 +29,14 @@ class PersonSerializer
         CarSerializer
       end
     end
+
+    has_many :polymorphic_assets, polymorphic: true, serializer: -> (object) do
+      if object.is_a?(House)
+        HouseSerializer
+      elsif object.is_a?(Car)
+        CarSerializer
+      end
+    end
```

Comment:
@doits I believe `polymorphic` is taking precedence, causing `serializer` to be disregarded.  Can you confirm that it behaves the same way when you provide `polymorphic` but remove `serializer`?

The problem you are showing is actually exactly why I needed #32: single table inheritance.  The `polymorphic` option will attempt to resolve `BigHouse` to `BigHouseSerializer`, which is undefined.  It has no way of knowing to discover `HouseSerializer`.  So it works in very narrow use cases, but that is why we have serializer procs now.

FWIW, if this was the limit of your use case, and all the object types were known up front, you could also define `polymorphic` as a Hash.  I'm sure someone designed it to help broaden the use cases, but it was only slightly better and extremely cumbersome.  The serializer proc will solve pretty much anything.

Reply:
Thanks for the info. And yes, it behaves the same if you remove `serializer` options.

Actually my use case was as simple as this: I am serializing a class `Address` that has a polymorphic relationship named `addressable`. So when defining my `AddressSerializer` I naturally assumed I should specify `polymorphic: true`, because, well, it is a polymorphic relationship ;-) It worked until an `addressable` was a STI class without an extra serializer, so I wanted to add the `serializer` option to explicitly state the serializer with a proc ... which didn't worked, and here we are. This is exactly what you say.

So for me I think the fix is simple (and I implemented it in my project already): Drop the `polymorphic` option and rely on `serializer` alone.

Just wanted to make you guys know there is something up here.